src/thor/costmatrix.cc

```diff
@@ -233,23 +234,31 @@ void CostMatrix::SourceToTarget(Api& request,
   }
 
   // Form the matrix PBF output
+  graph_tile_ptr tile;
   uint32_t count = 0;
   valhalla::Matrix& matrix = *request.mutable_matrix();
   reserve_pbf_arrays(matrix, best_connection_.size());
   for (const auto& connection : best_connection_) {
     uint32_t target_idx = count % target_location_list.size();
     uint32_t origin_idx = count / target_location_list.size();
-    float time = connection.cost.secs + .5f;
-    auto date_time = get_date_time(source_location_list[origin_idx].date_time(),
-                                   time_infos[origin_idx].timezone_index,
-                                   target_edgelabel_[target_idx].front().edgeid(), graphreader,
-                                   static_cast<uint64_t>(time));
+    float time = connection.cost.secs;
+    // no datetime for unfound connections
+    if (time < kMaxCost) {
```

Comment:
we don't add date_time when there's no connection, but we do add it for origin == target. in the serializer we put `null` when there was no connection